tests/proxy_overload_recovery.erl

```diff
@@ -0,0 +1,442 @@
+    %% If there are no thresholds there should never be an overload
+    eq(R, false);
+overloaded_post(#tstate{vnode_running = true}, _A,
+                {R, _VnodeQ = 0, _ProxyStatus}) ->
+    %% If the vnode is running, we have cleared queues so
+    %% should not be in overload.
+    eq(R, false);
+overloaded_post(#tstate{vnode_running = false,
+                        proxy_msgs = ProxyMsgs,
+                        threshold = Threshold}, _A,
+                {ResultOverload, _VnodeQ, _ProxyStatus}) ->
+    %% Either
+    %%   mailbox is completely an estimate based on proxy msgs
+    %%   or mailbox is a check + estimate since
+    %% The test cannot tell which as prior state may have left the counter
+    %% at any value between 0 and Interval.
+    %%
+    if
+        ProxyMsgs > Threshold ->  % definitely overloaded
+            eq(ResultOverload, true);
+        true ->
+            true % otherwise, not possible to check - vnode_resume gets us back
+    end.
+
+%%
+%% Test helpers
+%%
+
+%% Convert the test seed to request intervals, intervals and threshold
+%% if unset, remove from the app env so the proxy init auto-configures
+get_params(undefined) ->
+    {unset, unset, undefined};
+get_params({Interval, ThresholdDelta}) ->
+    {unset, % let the proxy allocate it
+     Interval,
+     Interval + ThresholdDelta};
+get_params({RequestInterval, IntervalDelta, ThresholdDelta}) ->
+    {RequestInterval,
+     RequestInterval + IntervalDelta,
+     RequestInterval + IntervalDelta + ThresholdDelta};
+get_params(Threshold) when Threshold > 5000 ->
+    {unset, unset, Threshold};
+get_params(Threshold) when Threshold < 5000 ->
+    %% Avoid annoying message about safety
+    {unset, Threshold div 2, Threshold}.
+
+
+%% Set (or unset) params
+prep_env(Var, unset) ->
+    application:unset_env(riak_core, Var, infinity);
+prep_env(Var, Val) ->
+    application:set_env(riak_core, Var, Val).
+
+%%
+%% Wait until all messages are drained by the Pid. No guarantees
+%% about future messages being sent.
+%%
+drain(Pid) ->
+    case erlang:process_info(Pid, message_queue_len) of
+        {message_queue_len, 0} ->
+            ok;
+        {message_queue_len, L} when L > 0 ->
+            timer:sleep(1), % give it a millisecond to drain
+            drain(Pid);
+        ER ->
+            ER
+    end.
+
+%% Return the length of the message queue (or crash if proc dead)
+msgq_len(Pid) ->
+    {message_queue_len, L} = process_info(Pid, message_queue_len),
+    L.
+
+%% Kill the process and wait for it to exit
+kill_and_wait(Pid) ->
+    Mref = erlang:monitor(process, Pid),
+    exit(Pid, kill),
+    receive
+        {'DOWN', Mref, _, _, _} ->
+            ok
+    end.
+
+%%
+%% Check if EQC is available on the local node, then add to remotes if needed
+%% This could be refactored out if many tests need it.
+setup_eqc(Nodes) ->
+    try
+        _ = eqc:reserved_until(), % will throw exception if problems
```

Comment:
The intent was that if eqc was not installed, or the license could not be reserved then we issue a warning saying the test did not actually run and pass it.

Unfortunately there's no way of skipping that I could work out looking at the r_ sources.